tutorials/auto-data-tokenize/index.md

```diff
@@ -0,0 +1,446 @@
+       "$.contacts[1].contact.number": "$.contacts.contact.number"   
+      }
+   }
+   ```
+
+## Prerequisites
+
+This tutorial assumes some familiarity with shell scripts and basic knowledge of Google Cloud.
+
+## Objectives
+
+1. Understand record sampling and identify sensitive columns using DLP.
+1. Use of symmetric encryption to tokenize data using KMS wrapped data-encryption key.
+
+## Costs
+
+This tutorial uses billable components of Google Cloud, including the following:
+
+* [Dataflow](https://cloud.google.com/dataflow/pricing)
+* [Cloud Storage](https://cloud.google.com/storage/pricing)
+* [Cloud Data Loss Prevention](https://cloud.google.com/dlp/pricing)
+* [Cloud KMS](https://cloud.google.com/kms/pricing)
+
+Use the [pricing calculator](https://cloud.google.com/products/calculator) to generate a cost estimate based on your
+projected usage.
+
+## Before you begin
+
+For this tutorial, you need a Google Cloud [project](https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy#projects). To make
+cleanup easiest at the end of the tutorial, we recommend that you create a new project for this tutorial.
+
+1. [Create a Google Cloud project](https://console.cloud.google.com/projectselector2/home/dashboard).
+1. Make sure that [billing is enabled](https://support.google.com/cloud/answer/6293499#enable-billing) for your Google
+   Cloud project.
+1. [Open Cloud Shell](https://console.cloud.google.com/?cloudshell=true).
+
+   At the bottom of the Cloud Console, a [Cloud Shell](https://cloud.google.com/shell/docs/features) session opens and
+   displays a command-line prompt. Cloud Shell is a shell environment with the Cloud SDK already installed, including
+   the [gcloud](https://cloud.google.com/sdk/gcloud/) command-line tool, and with values already set for your current
+   project. It can take a few seconds for the session to initialize.
+
+1. Enable APIs for Data Catalog, BigQuery, Pub/Sub, Dataflow, and Cloud Storage services with the following command:
+   ```shell script
+   gcloud services enable \
+   bigquery.googleapis.com \
+   storage.googleapis.com \
+   compute.googleapis.com \
+   dataproc.googleapis.com \
+   dataflow.googleapis.com \
+   dlp.googleapis.com \
+   kms.googleapis.com
+   ```
+
+## Setting up your environment
+
+1. In Cloud Shell, clone the source repository and go to the directory for this tutorial:
+   ```shell script
+   git clone https://github.com/GoogleCloudPlatform/auto-data-tokenize.git
+   cd auto-data-tokenize/
+   ```
+   > Use the following till this repo is open-sourced.
+   > ```shell script
+   > gcloud source repos clone automatic-dlp --project=google.com:anantd
+   > ```
+
+1. Use a text editor to modify the `set_variables.sh` file to set required environment variables.
+   ```shell script
+   # The GCP KMS resource URI
+   export MAIN_KMS_KEY_URI="gcp-kms://projects/auto-dlp/locations/asia-southeast1/keyRings/sym-wrapper-key/cryptoKeys/sym-1"
```

Comment:
should we move this down and use the ${PROJECT_ID} variable instead of hardcoding auto-dlp?